internal/jobobject/iocp.go

```diff
@@ -0,0 +1,105 @@
+package jobobject
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	ioInitOnce sync.Once
+	initIOErr  error
```

Comment:
Is this used any more?

Reply:
Yes, you need to open up the "too large to show" jobobject.go diff haha

Reply:
What's the point of this being a global? It's only used in one function and passed to the `do once` in that function. Why not use a local variable?

Reply:
If the IOCP can't be setup, we need to fail all future attempts to create a jobobject, not just the first. The case in Create currently actually will fail even if the caller doesn't specify that they want notifications so this https://github.com/microsoft/hcsshim/pull/877/files#diff-c85f523707b2a7adf91728349615e2d4c9ec4fc0b4c6f30d58d686aa63ed208dR140

should actually be 
```go
if initIOErr != nil && notifications {
```

Reply:
Wait I'm not sure I follow why we need to check that notifications is true?

Reply:
If they're not asking for notifications, we don't care if the iocp couldn't be setup as we're not going to use it anyways.